### ISSUE TITLE
Deprecate @ember/string usage.

### ISSUE THREAD

**Issue by lifeart** [emberjs/data#8711]
Since `ember-source` does not rely on `@ember/string` anymore.

We found usages of `@ember/string` in 
`@ember-data/adapter`, `@ember-data/debug`, `@ember-data/model`, `@ember-data/serializer`,
`@ember-data/store`, `ember-data` deps.

Code search: https://github.com/search?q=repo%3Aemberjs%2Fdata%20%22%40ember%2Fstring%22&type=code

It seems we could try to replace/omit it and completely deprecate `@ember/string`.

Related stuff: https://github.com/ember-cli/ember-resolver/issues/932, https://github.com/ember-cli/ember-resolver/pull/933

**Comment by runspired** [comment]
We do not use ember-source for it, we declare the peer-dependency. 

There isn't a good library out there that replaces it (or ember-inflector) I've investigated that a few times. I've been considering having EmberData ship its own string utils since long-term we only need `camelize` for the `REST` url builder, `underscore` for the `ActiveRecord` url builder, and `pluralize` for all builders. All other string functions *i think* end up unused except potentially for debug assertions once the deprecation for `type-matches-registry-name` is complete.

Closing since this work is already tracked in #8460 and #8147 and the RFC for strict types.
